File: examples/typescript/sdk-reference/observability/basic-usage.ts
import { DurableContext, withDurableExecution } from "@aws/durable-execution-sdk-js";

export const handler = withDurableExecution(async (event, context: DurableContext) => {
  context.logger.info("Starting workflow");

  const result = await context.step("process", async () => {
    return "done";
  });

  context.logger.info("Workflow complete", { result });
  return result;
});

File: examples/python/sdk-reference/observability/basic-usage.py
from aws_durable_execution_sdk_python import DurableContext, durable_execution


@durable_execution
def handler(event: dict, context: DurableContext):
    context.logger.info("Starting workflow")

    result = context.step(lambda ctx: "done")

    context.logger.info("Workflow complete", extra={"result": result})
    return result
